api-docs/client-interactions.md

```diff
@@ -0,0 +1,198 @@
+```
+
+### Response
+
+```json
+{
+  "interactions": [
+    {
+      "id": 1,
+      "title": "Nutrition Program - SNAP",
+      "interactionType": "by phone",
+      "notes": "Application successfully filled out. Decision pending.",
+      "startTime": "2019-05-06T05:00:00.000Z",
+      "endTime": "2019-05-06T06:00:00.000Z",
+      "location": null,
+      "referenceServices": ["Supermarket Visit"],
+      "canModify": true,
+      "isDeleted": false,
+      "createdBy": {
+        "userId": 1,
+        "firstName": "Joel",
+        "lastName": "Denning",
+        "fullName": "Joel Denning",
+        "timestamp": "2019-05-06T06:00:00.000Z"
+      }
+    },
+    {
+      "id": 2,
+      "title": "Financial Counseling - Financial Coach Referal",
+      "interactionType": "in person",
+      "notes": "Appointment setup with financial coach. Client would like to save up enough for a down payment on a car better suited for their work.",
+      "startTime": "2019-05-06T05:00:00.000Z",
+      "endTime": "2019-05-06T06:00:00.000Z",
+      "location": "CU Office",
+      "referenceServices": ["DACA", "Youth Group"],
+      "canModify": true,
+      "isDeleted": false,
+      "createdBy": {
+        "userId": 1,
+        "firstName": "Joel",
+        "lastName": "Denning",
+        "fullName": "Joel Denning",
+        "timestamp": "2019-05-06T06:00:00.000Z"
+      }
+    }
+  ]
+}
+```
+
+## Create a Client Interaction
+
+### Request
+
+```http
+POST /api/clients/:id/interactions
+```
+
+```json
+{
+  "title": "Financial Counseling - Financial Coach Referal",
+  "interactionType": "in person",
+  "notes": "Appointment setup with financial coach. Client would like to save up enough for a down payment on a car better suited for their work.",
+  "startTime": "2019-05-06T05:00:00.000Z",
+  "endTime": "2019-05-06T06:00:00.000Z",
+  "location": "CU Office",
+  "referenceServices": ["DACA", "Youth Group"],
+  "canModify": true,
+  "isDeleted": false,
+  "createdBy": {
+    "userId": 1,
+    "firstName": "Joel",
+    "lastName": "Denning",
+    "fullName": "Joel Denning",
+    "timestamp": "2019-05-06T06:00:00.000Z"
+  }
+}
+```
+
+**_Notes_**
+
+- `program` is an enum with the following possible string values `Preventive Health - VDS daily services`, `Preventive Health - VDS mobile`, `Preventive Health - VDS Sabath`, `Preventive Health - Mobile Health Services`, `Preventive Health - Other`, `Nutrition - Nutrition`, `Nutrition - Supermarket visit`, `Nutrition - Cooking class`, `Nutrition - SNAP`, `Immigration - Citizenship`, `Immigration - Family Petition`, `Immigration - DACA`, `Immigration - Other`, `Financial Counseling`, `Worker Rights`, `Youth Group`, `Community Organization - Volunteering`, `Community Organization - Promoter`, `Community Organization - Internship`, `Family Support Services - LDS Vouchers`, `Family Support Services - Faxes`, `Family Support Services - References outside of CU`, `Family Support Services - Other`
```

Comment:
`program` isn't in the API. I think that all we need is the `serviceId`. Since services are stored in the database, we should just reference them by their database primary key instead of by an enum.

Maybe in the POST api for creating interactions, we just specify the `serviceId`, but in the response bodies for all the APIs we specify both `serviceId` and `serviceName`?

Reply:
This still needs to be addressed.

Reply:
See question in comment above.